### PR TITLE
Enabling Kafka Trigger for secure Kafka clusters 

### PR DESCRIPTION
**Pull request by sushmithavangala**
PR for https://github.com/microsoft/k4apps/issues/523

For kafka function apps with bindings information with authentication , a TriggerAuthntication CRD is generated which keda uses to connect to secure cluster for scaling.

### CODE REVIEW

**Comment by sushmithavangala** [comment]
@pragnagopa  @TsuyoshiUshio @ahmelsayed I couldn't add any unit tests for KafkaTriggerKedaAuthProvider as it ends up calling the  K8SEDeploymentHelper.CreateTriggerAuthenticationRef(), which would run a build ctl command. I understand that we want to avoid running any build ctl command through test cases that would add/modify anything in the kubernetes cluster. Since K8SEDeploymentHelper is a static class we can't mock the CreateTriggerAuthenticationRef functionality. Let me know if there's any other way to add more unit tests.

File: Kudu.Core/Functions/KedaFunctionTriggerProvider.cs [hunk 1]
```diff
@@ -239,25 +253,25 @@ internal static string GetKedaTriggerType(string triggerType)
 
             switch (triggerType)
             {
-                case "queuetrigger":
+                case TriggerTypes.AzureStorageQueue:
```
Comment (TsuyoshiUshio) [inline Kudu.Core/Functions/KedaFunctionTriggerProvider.cs:256]: I'm wondering why you change it? I'd like to know the reason.

**Review by TsuyoshiUshio — COMMENTED** [review]
Only think I'm wondering is, is it possible to write unit test to validate your logic?
For example, you can find
https://github.com/Azure-App-Service/KuduLite/blob/feature/k8se/Kudu.Tests/Core/Function/KedaFunctionTriggersProviderTests.cs
https://github.com/Azure-App-Service/KuduLite/blob/feature/k8se/Kudu.Tests/Core/K8SE/K8SEDeploymentHelperTest.cs

File: Kudu.Core/Functions/KedaFunctionTriggerProvider.cs (hunk 1 above)
Comment (sushmithavangala) [reply Kudu.Core/Functions/KedaFunctionTriggerProvider.cs:256]: Modified the strings to use the constants consistently across the file, this way any changes to the constants can reflect in all the places.

Comment (TsuyoshiUshio) [reply Kudu.Core/Functions/KedaFunctionTriggerProvider.cs:256]: Thanks. That makes sense.

**Comment by sushmithavangala** [comment]
> Only think I'm wondering is, is it possible to write unit test to validate your logic? For example, you can find https://github.com/Azure-App-Service/KuduLite/blob/feature/k8se/Kudu.Tests/Core/Function/KedaFunctionTriggersProviderTests.cs https://github.com/Azure-App-Service/KuduLite/blob/feature/k8se/Kudu.Tests/Core/K8SE/K8SEDeploymentHelperTest.cs

**Comment by sushmithavangala** [comment]
@TsuyoshiUshio  referred https://github.com/TsuyoshiUshio/StaticMockMethodSample and added unit tests for KafkaTriggerKedaAuthProvider . KedaFunctionTriggerProvider and K8SEDeploymentHelper are themselves static classes, I haven't found a way to add test cases for methods insides these classes, which can abstract out the buid ctl part . Please let me know if you have an idea to add test cases to changes in these static classes.

File: Kudu.Core/K8SE/K8SEDeploymentHelper.cs [hunk 2]
```diff
@@ -124,6 +124,16 @@ public static void UpdateFunctionAppTriggers(string appName, IEnumerable<ScaleTr
             RunBuildCtlCommand(cmd.ToString(), "Updating function app triggers...");
         }
 
+        public static void CreateTriggerAuthenticationRef(string secretName, string authRefSecretKeyToParamMap, string appName)
+        {
+            var cmd = new StringBuilder();
+            BuildCtlArgumentsHelper.AddBuildCtlCommand(cmd, "createTriggerAuth");
+            BuildCtlArgumentsHelper.AddSecretName(cmd, secretName);
+            BuildCtlArgumentsHelper.AddAppNameArgument(cmd, appName);
+            BuildCtlArgumentsHelper.AddAuthRefSecretKeyToParamMap(cmd, authRefSecretKeyToParamMap);
```
Comment (TsuyoshiUshio) [inline Kudu.Core/K8SE/K8SEDeploymentHelper.cs:133]: Here

**Comment by TsuyoshiUshio** [comment]
Thank you for the update. for this PR, this test is enough. However, if you want to know how to add test for static method, you can extract the method of here with internal scope. https://github.com/Azure-App-Service/KuduLite/pull/230#discussion_r742507215 you can test the internal method. without executing actual command. This is one way. However, just FYI.

**Review by ahmelsayed — APPROVED** [review]

**Review by lpapudippu — APPROVED** [review]
Changes look good to me

**Review by TsuyoshiUshio — APPROVED** [review]
LGTM

**Pull request merged**